docs/1-introduction/1-1-getting-started/1-1-3-hello-triangle.md

```diff
@@ -1,11 +1,456 @@
+
+The input is 2, `float3` (vector of 3 float components).
+
+the first is the "position" field, which are coordinates ranging from -1.0 to 1.0, this range is called **NDC** or **Normalized Device Coordinates**, if the values are outside this range they are clipped, and we won't see them on screen.
+
+The second is the "color" field, which we also pass as the output of this stage onto the pixel shader.
+
+Notice how all our fields have a colon and some identifier attached to them, these are "semantics", they are basically a decorator of a certain field that tells us and D3D11 how we want to use this particular field. Semantics that are preceded by `SV` are called "system-value semantics", `SV_POSITION` for example means that the field `position` will be used by D3D11 as the actual output of the vertex shader.
+
+Then we have our `VSOutput`, which has our vertices in the first field `position` and our color in the second field `color`.
+
+Finally we have our main function, which takes in a single parameter which is our input in the form of `VSInput`, and returns our output in the form of a `VSOutput`. Since we don't do any processing, we simply make a new instance of `VSOutput`, initialize it all to 0 and forward our input position and color to the output.
+
+### Pixel Shader
+The **Pixel Shader** is the stage where we give the pixels on our render target color, it is invoked for each pixel that is covered by a triangle.
+
+We use this stage to apply most of our shading techniques, from basic lighting, to textures and shadows, all the way to physically based rendering.
+
+Since we did not specify any shader between the VS and the PS, our input here is the output of the VS, and the output is one or more render targets.
+
+Let's look at our Pixel Shader now:
+#### Main.ps.hlsl
+```hlsl
+struct PSInput
+{
+    float4 position: SV_POSITION;
+    float3 color: COLOR0;
+};
+
+struct PSOutput
+{
+    float4 color: SV_TARGET0;
+};
+
+PSOutput Main(PSInput input)
+{
+    PSOutput output = (PSOutput)0;
+    output.color = float4(input.color, 1.0);
+    return output;
+}
+```
+Here as well we have an input `PSInput`, and an output `PSOutput`.
+
+Since our input here is the VS's output, `PSInput` must match **exactly** the `VSOutput` in vertex shader, if the layout differ even a little, for example by moving `float3 color` above `float4 position`, the D3D11's debug layer will scream at us, and the shaders will fail to compile.
+
+Next we have our output, D3D11 gives us the possibility to write to multiple render targets, but we are not doing that, so we'll just be writing a `float4` as our output, which is an RGBA color.
+
+Notice how we have another semantic string attached to the `color` field, this semantic string specifies which render target we want to be writing to, the `0` after `SV_TARGET` is the index of our render target, in our case, we have only one so we write `SV_TARGET0` or `SV_TARGET`.
```

Comment:
Maybe add a paragraph, that most hardware these days supports up to 8 render targets which can be written to simultaneously (and comes in handy in scenarios like deferred rendering etc.. which we will cover in a later chapter)